File: library/src/test/java/org/crispy_fish/util/ResultConditions.java
package org.crispy_fish.util;

import org.assertj.core.api.Condition;
import org.assertj.core.condition.AllOf;
import org.assertj.core.util.Strings;
import org.coner.crispy_fish.domain.Result;
import org.coner.crispy_fish.domain.Numbers;

public class ResultConditions {

    public static Condition<Result> positionEquals(final int position) {
        return new Condition<>(result -> result.getPosition() == position, "result.position == %d", position);
    }

    public static Condition<Result> driverNumbersEqual(final Numbers numbers) {
        return new Condition<>(result -> result.getDriver().getNumbers().equals(numbers), "result.driver == %s", numbers.toString());
    }

    public static Condition<Result> driverNumbersEqual(final String classing, final String number) {
        Numbers numbers = new Numbers(classing, number);
        return driverNumbersEqual(numbers);
    }

    public static Condition<Result> driverNameNotNullOrEmpty() {
        return new Condition<>(result -> !Strings.isNullOrEmpty(result.getDriver().getName()), "result.driver is not null or empty");
    }

    public static Condition<Result> driverFinished(final int position, final String classing, final String number) {
        return AllOf.allOf(
                positionEquals(position),
                driverNumbersEqual(classing, number)
        );
    }
}

File: library/src/test/kotlin/org/coner/crispyfish/util/ResultConditions.kt
package org.coner.crispyfish.util

import org.assertj.core.api.Condition
import org.assertj.core.condition.AllOf
import org.assertj.core.util.Strings
import org.coner.crispyfish.model.Registration
import org.coner.crispyfish.model.Result
import java.util.function.Predicate

object ResultConditions {

    fun positionEquals(position: Int) = Condition<Result>(
                Predicate{ result -> result.position == position },
                "result.position == %d",
                position
    )

    fun driverNumbersEqual(category: String?, handicap: String, number: String) = Condition<Result>(
            Predicate { result ->
                (result.driver?.category?.abbreviation ?: "").equals(category, ignoreCase = true)
                && result.driver?.handicap?.abbreviation.equals(handicap, ignoreCase = true)
                && result.driver?.number == number
            },
            "result.driver == %s",
            "${category ?: ""}$handicap $number"
    )

    fun driverNameNotNullOrEmpty(): Condition<Result> {
        return Condition(Predicate{ result ->
            !result.driver?.firstName.isNullOrBlank()
            && !result.driver?.lastName.isNullOrBlank()
        }, "result.driver first and last last names are not null or blank")
    }

    fun driverFinished(position: Int, category: String?, handicap: String, number: String): Condition<Result> {
        return AllOf.allOf(
                positionEquals(position),
                driverNumbersEqual(category, handicap, number)
        )
    }
}
